src/widgets/mainwindow.cpp

```diff
@@ -485,3 +536,63 @@ void MainWindow::setStayOnTop(bool p_enabled)
         show();
     }
 }
+
+void MainWindow::initSystemTrayIcon(){
+    QMenu *menu = new QMenu(this);
+    QAction *showMainWindowAct = menu->addAction(tr("Show VNote"));
+    connect(showMainWindowAct, &QAction::triggered,
+            this, &MainWindow::show);
+
+    QAction *exitAct = menu->addAction(tr("Quit"));
+    connect(exitAct, &QAction::triggered,
+            this, [this](){
+                this->m_requestQuit = 1;
+                this->close();
+                });
+
+    QIcon sysIcon(":/vnotex/data/core/logo/vnote.png");
+
+#if defined(Q_OS_MACOS) || defined(Q_OS_MAC)
+    sysIcon.setIsMask(true);
+#endif
+
+    m_trayIcon = new QSystemTrayIcon(sysIcon, this);
+    m_trayIcon->setToolTip(tr("VNote"));
+    m_trayIcon->setContextMenu(menu);
+
+    connect(m_trayIcon, &QSystemTrayIcon::activated,
+            this, [this](QSystemTrayIcon::ActivationReason p_reason){
+#if !defined(Q_OS_MACOS) && !defined(Q_OS_MAC)
+                if (p_reason == QSystemTrayIcon::Trigger) {
+                    this->show();
+                }
+#endif
+            });
+
+    m_trayIcon->show();
+}
+
+const int MainWindow::c_sharedMemTimerInterval = 1000;
+void MainWindow::initSharedMemoryWatcher(){
```

Comment:
Could you please remove the single-instance part logics in this PR? One reasone is that one PR had better do one thing. The other reason is that there is a bug in the shared memory on macOS, I want to refactor it to use HTTP port.